### ISSUE TITLE
Broken .dll for Windows 10 x64

### ISSUE THREAD

**Issue by gingeralesy** [cbaggers/cl-soil#4]
Had an issue where cl-soil would read a PNG file and try to return a pointer to the data. This cffi:pointer was broken and caused a crash when freed. Skipping freeing the pointer it turned out that the data was also invalid and would end up jamming up entire process until killed.

The solution to this was for me to compile my own copy of _lib/local_x64_libSOIL.dll_. After that everything worked marvelously once again.

I used mingw64-gcc to make the compile. Checking the dependancy libraries for the .dll gives the same results for both the working and broken copy, so it likely is not a linking error.

Probably unrelated to this, but I had a different error on top of the one that caused the crash and it went away once I updated my nvidia drivers.

**Comment by Shinmera** [comment]
The local_x86_libSOIL.dll library is actually also broken because it is a 64x one, not a 32x one.

**Comment by cbaggers** [comment]
Thanks to both of you! I've been using appveyor to build the library so I guess I've screwed something up there. I'll dig into it soon

**Comment by Shinmera** [comment]
The most reliable way to build unix-y libraries for Windows without dependencies or other cruft is to use [msys2](http://www.msys2.org/) and its MinGW64/32 consoles and package sets.

**Comment by cbaggers** [comment]
Thanks. As I really don't want to have a machine I own doing this, but I'll look into this on appveyor if needed

**Comment by Shinmera** [comment]
I just use a Windows 7 VM.

**Comment by cbaggers** [comment]
Indeed, but that is still a machine that needs a windows license & me to set it up and keep it up to date. By using CI I get builds tested and get to push the resulting libs back to github which is super handy for me. Also it's free for opensource projects.

Thanks for the tips though
